snap/info.go

```diff
@@ -743,19 +754,22 @@ func lookupAttr(attrs map[string]interface{}, path string) (interface{}, bool) {
 func getAttribute(snapName string, ifaceName string, attrs map[string]interface{}, key string, val interface{}) error {
 	v, ok := lookupAttr(attrs, key)
 	if !ok {
-		return fmt.Errorf("snap %q does not have attribute %q for interface %q", snapName, key, ifaceName)
+		return AttributeNotFoundError{
+			fmt.Errorf("snap %q does not have attribute %q for interface %q", snapName, key, ifaceName),
+		}
 	}
 
 	rt := reflect.TypeOf(val)
```

Comment:
it seems there is more code that can be shared between the functions than  just TryConvert. A more general SetValueFromAttribute or something?

Reply:
Yes, that makes sense, I'll try.

Reply:
Done